External/FEXCore/Source/Interface/Context/Context.h

```diff
@@ -343,6 +346,9 @@ namespace FEXCore::Context {
     bool StartPaused = false;
     bool IsMemoryShared = false;
     FEX_CONFIG_OPT(AppFilename, APP_FILENAME);
+    
+    std::shared_mutex CustomIRMutex;
+    std::unordered_map<uint64_t, std::tuple<std::function<void(uintptr_t Entrypoint, FEXCore::IR::IREmitter *)>, void *, void *>> CustomIRHandlers;
```

Comment:
I'd much prefer a struct here rather than a tuple, but I trust we're providing a big enough foot-gun that the semantics don't matter here.
Opaque interfaces are fine.

Reply:
I'll unlazy and add struct + typedef for the function type as well. I guess we should also do that in other places that magical tuples are used?